data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/client/PeerForwarderClient.java

```diff
@@ -57,17 +58,18 @@ public AggregatedHttpResponse serializeRecordsAndSendHttpRequest(
 
         final WebClient client = peerClientPool.getClient(ipAddress);
 
-        final String serializedJsonString = getSerializedJsonString(records, pluginId);
-        if (serializedJsonString == null) {
+        final Optional<String> serializedJsonString = getSerializedJsonString(records, pluginId);
+        if (!serializedJsonString.isPresent()) {
```

Comment:
Nit: Take it or leave it. You could simplify L62-L69 into 
```
return serializedJsonString.map(value -> {
   final CompletableFuture<AggregatedHttpResponse> aggregatedHttpResponseCompletableFuture = processHttpRequest(client, value);
   return getAggregatedHttpResponse(aggregatedHttpResponseCompletableFuture);
})
.orElse(AggregatedHttpResponse.of(HttpStatus.BAD_REQUEST))
```

Reply:
Took it.